### PR TITLE
Internal: Add a badge for cloud experiments [HTS-336]

### PR DESCRIPTION
**Pull request by mark-elementor**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: core/experiments/manager.php [hunk 1]
```diff
@@ -740,6 +781,8 @@ private function validate_dependency( array $feature, $new_state ) {
 
 		$feature_option_key = $this->get_feature_option_key( $feature['name'] );
 
+		error_log( 'new state: ' . $new_state );
+
```
Comment (matipojo) [inline core/experiments/manager.php:785]: ?

File: core/experiments/manager.php [hunk 2]
```diff
@@ -149,6 +152,44 @@ public function add_feature( array $options ) {
...
+		foreach ( $tags as &$tag ) {
+			if ( empty( $tag['label'] ) ) {
+				unset( $tag );
+			}
+
+			$tag = $this->merge_properties( $default_tag, $tag, $allowed_tag_properties );
+		}
+
+		return $tags;
```
Comment (matipojo) [inline core/experiments/manager.php:190]: Please add it to the usage data + schema

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/experiments/manager.php (hunk 1 above)
Comment (mark-elementor) [reply core/experiments/manager.php:785]: Oh oops. Should not be here! Removed :)

File: core/experiments/manager.php (hunk 2 above)
Comment (mark-elementor) [reply core/experiments/manager.php:191]: Added

File: core/experiments/manager.php [hunk 3]
```diff
@@ -149,6 +153,72 @@ public function add_feature( array $options ) {
...
+	 *
+	 * @param array $experimental_data
+	 *
+	 * @return array
+	 */
+	private function unify_feature_tags( array $experimental_data ) : array {
+		foreach ( [ 'tag', 'tags' ] as $key ) {
+			if ( ! empty( $experimental_data[ $key ] ) ) {
+				$experimental_data[ $key ] = $this->format_feature_tags( $experimental_data[ $key ] );
+			}
```
Comment (nivnoimanelementor) [inline core/experiments/manager.php:167]:
```suggestion
			if ( empty( $experimental_data[ $key ] ) ) {
				continue;
			}
			$experimental_data[ $key ] = $this->format_feature_tags( $experimental_data[ $key ] );
```

Comment (nivnoimanelementor) [inline core/experiments/manager.php:167]:
```suggestion
			if ( empty( $experimental_data[ $key ] ) ) {
				continue;
			}
			$experimental_data[ $key ] = $this->format_feature_tags( $experimental_data[ $key ] );
```

**Review by nivnoimanelementor — CHANGES_REQUESTED** [review]
Great, just one little change please 🙏

**Review by nivnoimanelementor — DISMISSED** [review]

File: core/experiments/manager.php (hunk 3 above)
Comment (mark-elementor) [reply core/experiments/manager.php]: Done

File: tests/phpunit/elementor/core/experiments/test-manager.php [hunk 4]
```diff
@@ -58,6 +59,180 @@ public function test_add_feature() {
 		$this->assertEquals( null, $re_added_feature );
 	}
 
+	public function test_add_feature__with_tags() {
```
Comment (matipojo) [inline tests/phpunit/elementor/core/experiments/test-manager.php:62]: Try to use the `@dataProvider` approach

Example: https://github.com/elementor/elementor/blob/5eb5cf28f37fbf2ad35cc68319ff830ca9249750/tests/phpunit/elementor/app/import-export/test-module.php#L163

https://github.com/elementor/elementor/blob/5eb5cf28f37fbf2ad35cc68319ff830ca9249750/tests/phpunit/elementor/app/import-export/test-module.php#L193

**Review by matipojo — CHANGES_REQUESTED** [review]

**Review by nivnoimanelementor — DISMISSED** [review]

**Review by nivnoimanelementor — APPROVED** [review]

**Pull request merged**